### PR TITLE
update repository, snapshots, split stream and upcast docs

### LINKED ISSUE #481: Prepare docs for 3.0.0
- [x] readme https://github.com/patchlevel/event-sourcing/pull/440
- [x] aggregate https://github.com/patchlevel/event-sourcing/pull/483
- [x] aggregate id https://github.com/patchlevel/event-sourcing/pull/483
- [x] cli https://github.com/patchlevel/event-sourcing/pull/505
- [x] clock https://github.com/patchlevel/event-sourcing/pull/483
- [x] message https://github.com/patchlevel/event-sourcing/pull/564
- [x] event bus https://github.com/patchlevel/event-sourcing/pull/483
- [x] events https://github.com/patchlevel/event-sourcing/pull/483
- [x] getting started https://github.com/patchlevel/event-sourcing/pull/440
- [x] index https://github.com/patchlevel/event-sourcing/pull/440
- [x] message decorator https://github.com/patchlevel/event-sourcing/pull/483
- [x] normalizer https://github.com/patchlevel/event-sourcing/pull/510
- [x] projection https://github.com/patchlevel/event-sourcing/pull/503
- [x] repository https://github.com/patchlevel/event-sourcing/pull/507
- [x] snapshots https://github.com/patchlevel/event-sourcing/pull/507
- [x] splitstream https://github.com/patchlevel/event-sourcing/pull/507
- [x] store https://github.com/patchlevel/event-sourcing/pull/505
- [x] tests https://github.com/patchlevel/event-sourcing/pull/507
- [x] upcasting https://github.com/patchlevel/event-sourcing/pull/507

### PR DESCRIPTION
**Pull request by DavidBadura**

### CODE REVIEW

File: docs/pages/snapshots.md [hunk 1]
```diff
@@ -5,12 +5,17 @@ This is not a problem if there are a few hundred.
 But if the number gets bigger at some point, then loading and rebuilding can become slow.
 The `snapshot` system can be used to control this.
 
-Normally, the events are all executed again on the aggregate in order to rebuild the current state.
+!!! note
+
+    In oure benchmarks we can load 10 000 events for one aggregate in 50ms.
```
Comment (DanielBadura) [inline docs/pages/snapshots.md]:
```suggestion
    In our benchmarks we can load 10 000 events for one aggregate in 50ms.
```

Generally im not sure if this note here is *really* helpful. Maybe add a separate section about performance with more context etc?

File: docs/pages/split_stream.md [hunk 2]
```diff
@@ -1,45 +1,86 @@
...
+So you can create projections which are based on the full event stream.
 
-To use this feature you need to add the `SplitStreamDecorator`. You will also need events which will trigger this
-action. For that you can use the `#[SplitStream]` attribute. We decided that we are not literallty splitting the stream,
-instead we are marking all past events as archived as soon as this event is saved. Then the past events will not be
-loaded anymore for building the aggregate. This means that all needed data has to be present in these events which
-should trigger the event split.
+!!! tip
+
+    To split the stream is often a better solution than using snapshots.
```
Comment (DanielBadura) [inline docs/pages/split_stream.md]: I'm not sure if this statement is really correct. Performance wise Snapshotting is better - but not related to any business requirements.

File: docs/pages/snapshots.md (hunk 1 above)
Comment (DavidBadura) [reply docs/pages/snapshots.md]: I revised the text to make it clear what I wanted to say.

File: docs/pages/split_stream.md (hunk 2 above)
Comment (DavidBadura) [reply docs/pages/split_stream.md]: removed

**Pull request merged**